src/ipset/ipset_hash.c

```diff
@@ -167,23 +166,26 @@ hash_del(struct ipset *set, void *value, uint16_t flag)
 static inline int
 do_test(struct ipset *set, struct hash_type *htype, void *elem)
 {
+    int res;
     uint32_t key;
     struct hash_entry *hnode;
     struct list_head *head ;
 
     key = do(hash, elem, set->hash_len, htype->mask);
     head = &htype->htable[key];
     list_for_each_entry(hnode, head, list) {
-        if (do(compare, elem, hnode->elem))
-            return 1;
+        res = do(compare, elem, hnode->elem);
+        if (res == COMPARE_EQUAL_ACCEPT ||
```

Comment:
Is it ok to use `if (res != COMPARE_INEQUAL)`

Reply:
yes. but it's easier to understand in this way.